block-switch/block-switch-core/src/main/resources/blockSwitch.js

```diff
@@ -11,6 +13,13 @@ function addBlockSwitches() {
 		switchItem.content.addClass('hidden');
 		findPrimary(secondary).append(switchItem.content);
 		secondary.remove();
+		var groupName = primary.find('div.switch--item').text();
```

Comment:
Thanks for the updates.

I don't think the name of the primary item alone can be used as the group name. For example, if some documentation has Java, Kotlin, and XML configuration options in some places but only Java and Kotlin in others, selecting XML will break those that only offer Java and Kotlin. This is what I was trying to get at in an earlier comment:

> I think we need to iterate over all of the switches and group them based on the choices that they offer. A value then needs to be stored for each distinct set of choices.

After some more thought, I don't think we need to iterate, but we do need to consider the text of the primary node and all of the secondary nodes when creating the identifier that's used to remember the user's selection. For example if `Java`, `Kotlin`, and `XML` are the choices the identifier could be `java-kotlin-xml`. If `Java` and `Kotlin` are the choices, the identifier could be `java-kotlin`. I don't think the ordering of the choices should matter so `Java`, `Kotlin`, and `XML` and `Java`, `XML`, and `Kotlin` would both produce an identifier of `java-kotlin-xml`. I've lower-cased the text when creating the identifiers as I don't think the case should matter either, i.e. we wouldn't want `XML` and `Xml` to be considered separately.

Do you have time to update the PR along these lines?

Reply:
Yup. I'll continue to plunk away at it. Thanks, Andy.

Reply:
Why are we using "primary" and "secondary"? What happens when we want three choices? I think this would be better for authors and easier for coding this JavaScript if it were a single word for all the values - "switchChoice" or something like that. Then the trick will be grouping them, but we can look for intervening content. Can we do something like that?

Reply:
> What happens when we want three choices?

As [described in the README](https://github.com/spring-io/spring-asciidoctor-extensions#block-switch), you must have one `primary` block and one or more `secondary` blocks.

>  I think this would be better for authors and easier for coding this JavaScript if it were a single word for all the values - "choice" or something like that.

The use of two different terms makes it possible to identify where each set of choices begins. `primary` indicates the start of a set of choices and every subsequent `secondary` can be associated with the preceding `primary` until the next `primary` is encountered. In terms of the JavaScript, I'm not sure that it would be possible to group the choices if the same term was used for all values. I'm also not sure that it would be better for authors as it would be harder to see the groupings in the source as well.

> Can we do something like that?

I'm not opposed to changing the terms if we think that the names could be better. However, it would be a breaking change so we should only do it if there's strong justification to do so. It should also, ideally, be done separately to this change as the two need not be related.